### PR TITLE
[Dygraph]Add multiclass_nms,anchor_generator,prior_box in dygraph

### PR DESCRIPTION
**Pull request by yghstill**
as title.

### CODE REVIEW

File: ppdet/modeling/tests/test_ops.py [hunk 1]
```diff
@@ -459,5 +454,184 @@ def test_yolo_box_error(self):
...
+
+        self.assertTrue(np.array_equal(anchor_np, anchor_dy_np))
+        self.assertTrue(np.array_equal(var_np, var_dy_np))
+
+
+class TestMulticlassNms(LayerTest):
+    def test_prior_box(self):
+        boxes_np = np.random.rand(81, 4).astype('float32')
+        scores_np = np.random.rand(81).astype('float32')
+        rois_num_np = np.array([4, 6]).astype('int32')
```
Comment (wangxinxin08) [inline ppdet/modeling/tests/test_ops.py:558]: rois_num的和应该等于81

Comment (yghstill) [reply ppdet/modeling/tests/test_ops.py]: Done, thanks

File: ppdet/modeling/ops.py [hunk 2]
```diff
@@ -669,3 +672,411 @@ def yolo_box(
...
+              min_sizes,
+              max_sizes=None,
+              aspect_ratios=[1.],
+              variance=[0.1, 0.1, 0.2, 0.2],
+              flip=False,
+              clip=False,
+              steps=[0.0, 0.0],
+              offset=0.5,
+              name=None,
+              min_max_aspect_ratios_order=False):
```
Comment (jerrywgz) [inline ppdet/modeling/ops.py:688]: name放在最后吧

File: ppdet/modeling/ops.py [hunk 3]
```diff
@@ -669,3 +672,411 @@ def yolo_box(
...
+    """
+
+    This op generates prior boxes for SSD(Single Shot MultiBox Detector) algorithm.
+    Each position of the input produce N prior boxes, N is determined by
+    the count of min_sizes, max_sizes and aspect_ratios, The size of the
+    box is in range(min_size, max_size) interval, which is generated in
+    sequence according to the aspect_ratios.
+
+    Parameters:
+       input(Variable): 4-D tensor(NCHW), the data type should be float32 or float64.
```
Comment (jerrywgz) [inline ppdet/modeling/ops.py:698]: Variable -> Tensor

File: ppdet/modeling/ops.py [hunk 4]
```diff
@@ -669,3 +672,411 @@ def yolo_box(
...
+            Caffe. Please note, this order affects the weights order of
+            convolution layer followed by and does not affect the final
+            detection results. Default: False.
+       name(str, optional): The default value is None.  Normally there is no need for 
+            user to set this property. For more information, please refer to :ref:`api_guide_Name`
+
+    Returns:
+        Tuple: A tuple with two Variable (boxes, variances)
+
+        boxes(Variable): the output prior boxes of PriorBox.
```
Comment (jerrywgz) [inline ppdet/modeling/ops.py:726]: Variable -> Tensor

File: ppdet/modeling/ops.py [hunk 5]
```diff
@@ -669,3 +672,411 @@ def yolo_box(
...
+                     name=None):
+    """
+
+    This op generate anchors for Faster RCNN algorithm.
+    Each position of the input produce N anchors, N =
+    size(anchor_sizes) * size(aspect_ratios). The order of generated anchors
+    is firstly aspect_ratios loop then anchor_sizes loop.
+
+    Args:
+       input(Variable): 4-D Tensor with shape [N,C,H,W]. The input feature map.
```
Comment (jerrywgz) [inline ppdet/modeling/ops.py:835]: Tensor

File: ppdet/modeling/ops.py [hunk 6]
```diff
@@ -669,3 +672,411 @@ def yolo_box(
...
+                                            score_threshold=0.5,
+                                            nms_top_k=400,
+                                            nms_threshold=0.3,
+                                            keep_top_k=200,
+                                            normalized=False,
+                                            return_index=True)
+    """
+    helper = LayerHelper('multiclass_nms3', **locals())
+
+    if in_dygraph_mode():
```
Comment (jerrywgz) [inline ppdet/modeling/ops.py:1039]: 动态图里增加return_index的逻辑

File: ppdet/modeling/ops.py [hunk 7]
```diff
@@ -669,3 +672,411 @@ def yolo_box(
...
+        assert rois_num is not None, "rois_num should not be None in dygraph mode."
+        attrs = ('background_label', background_label, 'score_threshold',
+                 score_threshold, 'nms_top_k', nms_top_k, 'nms_threshold',
+                 nms_threshold, 'keep_top_k', keep_top_k, 'nms_eta', nms_eta,
+                 'normalized', normalized)
+        output, index, nms_rois_num = core.ops.multiclass_nms(bboxes, scores,
```
Comment (jerrywgz) [inline ppdet/modeling/ops.py:1045]: multiclass_nms3

File: ppdet/modeling/ops.py (hunk 6 above)
Comment (yghstill) [reply ppdet/modeling/ops.py:1033]: Done, thanks

File: ppdet/modeling/ops.py (hunk 2 above)
Comment (yghstill) [reply ppdet/modeling/ops.py]: Done

File: ppdet/modeling/ops.py (hunk 7 above)
Comment (yghstill) [reply ppdet/modeling/ops.py]: Done

**Pull request closed without merging**